src/librustc_mir/borrow_check/universal_regions.rs

```diff
@@ -462,13 +503,14 @@ impl<'cx, 'tcx> UniversalRegionsBuilder<'cx, 'tcx> {
             defining_ty,
             unnormalized_output_ty,
             unnormalized_input_tys,
-            yield_ty: yield_ty,
+            yield_ty,
+            diagnostic_dup_regions: dup_regions,
```

Comment:
```suggestion
            diagnostic_dup_regions,
```
Would be clearer to just use `diagnostic_dup_regions` locally as well as it clarifies the purpose (and what it *isn't* for) immediately. Same idea in `fn defining_ty` below.